project/src/main/webapp/script.js

```diff
@@ -0,0 +1,85 @@
+var map;
+var autocomplete;
+var photoURLS; 
+var image;
+
+function execute() {
+    initMap();
+    search();
+    getCatalog();
+}
+
+function initMap() {
+    map = new google.maps.Map(document.getElementById("map"), {
+        center: { lat: -33.866, lng: 151.196 },
+        zoom: 15
+    });
+}
+
+function search() {
+    // autocompleted suggestions of user searches
+    var input = document.getElementById("pac-input");
+    autocomplete = new google.maps.places.Autocomplete(input);
+    autocomplete.setFields(["place_id", "geometry"]);
+}
+
+function getCatalog() {
```

Comment:
I imagined that this will take in the the placeId as a parameter, but if had a different idea for implementing this please let me know. As of now when a location is clicked I call showCatalogue(placeId)